epc_data_analysis/pipeline/easy_plotting.py

```diff
@@ -0,0 +1,606 @@
+# File: getters/easy_plotting.py
+"""Functions to help plot and analyse dataframe.
+
+Created May 2021
+@author: Julia Suter
+Last updated on 13/07/2021
+"""
+
+# ---------------------------------------------------------------------------------
+
+# Imports
+import pandas as pd
+import matplotlib.pyplot as plt
+from matplotlib.gridspec import GridSpec
+
+import re
+
+from epc_data_analysis import get_yaml_config, Path, PROJECT_DIR
+from typing import Union
+
+import warnings
+
+warnings.simplefilter("ignore", category=UserWarning)
+
+# ---------------------------------------------------------------------------------
+
+# Load config file
+epc_data_config = get_yaml_config(
+    Path(str(PROJECT_DIR) + "/epc_data_analysis/config/base.yaml")
+)
+
+# Where to store figures
+FIG_PATH = str(PROJECT_DIR) + epc_data_config["FIGURE_PATH"]
+
+
+def save_figure(plt, save_filename, plot_title=None, file_extension=".png", dpi=500):
+    """Create filename and save figure.
+
+    Parameters
+    ----------
+
+    plt : matplotlib.pyplot
+        Plot to save.
+
+    save_filename : str, None
+        Save plot under this filename, automatically add ".png"
+        If None, do not save.
+        If "" or "auto", automatically generate filename.
+
+    plot_title: str, None, default=None
+        Use plot title to generate filename.
+        If None, use "figure" as filename.
+
+    file_extension: str, default=".png"
+        File extension, file format to save.
+
+    dpi: int, default=500
+        Dots per inches (dpi) determines how many pixels the figure comprises.
+
+
+    Return
+    ----------
+        None"""
+
+    # Tight layout
+    plt.tight_layout()
+
+    # Only save plot if required
+    if save_filename is not None:
+        if save_filename == "" or save_filename == "auto":
+
+            # Automatically enerate filename
+            if plot_title is not None:
+                save_filename = plot_title
+                save_filename = re.sub(" ", "_", save_filename)
```

Comment:
Is there a benefit to being able to having the choice between plot_title and save_filename?

Reply:
Maybe, but so far I never needed that distinction. I removed save_filename for now.